### ISSUE TITLE
[BUG] TypingTimeout appears to have no effect.

### ISSUE THREAD

**Issue by galenhuntington** [TiddlyWiki/TiddlyWiki5#5505]
I am new to this project, so I may be missing something, but I have spent hours on this to no avail.  The setting `$:/config/Drafts/TypingTimeout` [says](https://tiddlywiki.com/#Hidden%20Setting%3A%20Typing%20Refresh%20Delay)

> The timeout can now be changed by changing this value (in milliseconds)

I get noticeable fits and starts while typing due to the constant sync actions, so I tried increasing this.  It does not appear to have any effect.

**To Reproduce**
1. Change `$:/config/Drafts/TypingTimeout` to a high value (say 60000).
2. Start typing in a draft text box.
3. There is still high-frequency sync and save.

**Desktop**
 - Linux/Ubuntu
 - Firefox 85

**Additional context**

The timeout appears to be implemented in `core/modules/startup/render.js`.  With suitable logging, I can see the `refresh()` function is in fact delayed appropriately.  But the updates still happen; I can see the messages in the logs such as "Dispatching 'save' task", and the draft file is visible updated.  My tentative surmise is that some other change event must make it moot.  That is, numerous events fire on every keystroke (executing a lot of code), and somewhere the update happens anyway, so the delay is irrelevant.

I have tried increasing `taskTimerInterval`, but that doesn't do much either.  Increasing the throttle interval (using `$:/config/SyncThrottleInterval`, though it seems to require a server restart) at least slows down saving, which helps somewhat.

That's as far as I've gotten.

**Comment by Jermolene** [comment]
Hi @galenhuntington assuming you're running under Node.js, there are several synchronisation steps between typing a character in a draft and the `.tid` file being saved to the wiki folder:

* `$:/config/Drafts/TypingTimeout` determines how long must elapse after the last character typed in a draft before the render tree to be refreshed
* `$:/config/SyncThrottleInterval` determines how long must elapse after the last modification to a tiddler in the browser before it is synced from the browser to the syncer
* `$:/config/SyncPollingInterval` goes the other way, determining how often the browser polls the server for changes

To determine whether it's the refresh mechanism or the syncer mechanism that's causing your issue, you could take a snapshot of the wiki using the "save changes" button in the browser, and then open it and try editing something to see if it is subject to the same problem.

**Comment by galenhuntington** [comment]
Yes I'm running `tiddlywiki` under Node.  I've looked around and can't find a "save changes" button, although I see the text for such a button in the repo.

But, just to explain, I added a `console.log("refreshing")` to the `refresh()` function in `core/modules/startup/render.js`.  With `TypingTimeout` set to a high number, I see console output like this whenever I type something into the draft textarea:

```
<empty string> syncer-browser-tiddlyweb: Checking dirty status <empty string>
<empty string> syncer-browser-tiddlyweb: Dispatching 'save' task: Draft of 'New Tiddler' <empty string>
<empty string> syncer-browser-tiddlyweb: Checking dirty status <empty string>
```

("Checking dirty status" is repeated with every keypress, and also just repeated all the time anyway.)

At that point I do a `cat tiddlers/Draft\ of\ \'New\ Tiddler\'.tid` and see that what I've typed has been saved.  Only much later do I see `refreshing` logged, but no further changes (including to `modified:`).  So I'm not sure what that refresh is doing.

That is, what I've typed into the textarea is at least being synched to the server quickly, and I would assume it would go through whatever internal data structure the code running in the browser uses, which makes me think typing characters immediately causes the draft to be saved internally, which is why it seems `refresh()` and thus `TypingTimeout` don't matter.  (Or is the textarea contents directly saved to the server on sync?  That would still make the timeout setting nearly immaterial.)

Tangentially, I also noticed that with the `SyncThrottleInterval` set higher, the first keystroke after a period of inactivity still gets saved to the server immediately, although subsequent ones are delayed.  I guess that's because throttling only applies between saves.

**Comment by galenhuntington** [comment]
Okay, so I do see that if I have the preview window open, then the `TypingTimeout` refresh is when it updates.  So setting it higher delays that, which is something.  But it's not preventing the sync to server.

**Comment by pmario** [comment]
@galenhuntington ... Hi, Are you still with us? Did the SyncThrottleInterval solve your problem?
Can we close this issue?

**Comment by galenhuntington** [comment]
I haven't noticed the problem with the jumpy fits and starts while typing in a long time.  So that is somehow fixed, in that it seems the syncs don't stall character rendering.  I do still observe this:

> Tangentially, I also noticed that with the `SyncThrottleInterval` set higher, the first keystroke after a period of inactivity still gets saved to the server immediately, although subsequent ones are delayed.  I guess that's because throttling only applies between saves.

But that's not too important.  So sure let's close.
